### ISSUE TITLE
Update/Clarify rmq.all versus rmq(view).all

### ISSUE THREAD

**Issue by squidpunch** [infinitered/rmq#274]
when I build a new app `rmq create blah` then add `rmq.append(UIView).tag(:stuff).append(UIButton)`

```
(main)> rmq.all.count
=> 3
(main)> rmq(:stuff).count
=> 1
(main)> rmq(:stuff).all.count
# I really wouldnt expect "all" to go outside of the filtered scope but here is the result
=> 3
(main)>
```

The view has the following appended: 
- 1 UILabel (from the template)
- 1 UIView (tagged :stuff)`
- 1 UIButton (inside the UIView appended)

I would expect this to actually report as:
```
(main)> rmq.all.count
=> 3 - because there are 3 views appended to the root - and this is correct today
(main)> rmq(:stuff).count
=> 1 - also correct, there is only one tagged with that tag/style
(main)> rmq(:stuff).all.count
=> 1 - the view that i selected only contains 1 item (or should it be 2 to include self by default?)
(main)>
```
we can get the expected results by using `rmq(:stuff).find.count` but `.all` in a filtered context is confusing to me to return all ignoring the filter

what do others think?

**Comment by twerth** [comment]
Yes, we should change this IMO.

**Comment by GantMan** [comment]
the idea of saying `.all` once inside of a filtered scope should (yes) behave the way you mentioned, but it should also print a warning that it's redundant.   `rmq(:some_view).all == rmq(:some_view)`.   It doesn't read as explicit to me, it reads as "why did they do this?"

**Comment by twerth** [comment]
I don't understand @GantMan? It's not the same.

**Comment by GantMan** [comment]
I'm saying they would be the same once the scope is narrowed

**Comment by twerth** [comment]
I actually wouldn't. His example makes it appear like it would. `.all` would be all children, grandchildren, etc. I would be the same as `.find(UIView)`

**Comment by GantMan** [comment]
O!  Makes sense.  I didn't think of nested views.  good call.

**Comment by squidpunch** [comment]
@GantMan yeah this is totally related to nested views, thats my problem, not all the selection, but all the views from the selection because I want whats contained in below the filter, not just the match to the query.

I'll work out a PR to change this.  I want to review with you @twerth at some point because we talked about `all` including the root or not - and want to make sure I understand that correctly before working on this.

in the template app (which just pushes a label onto the view) `rmq.all` only includes 1 view (the label) so when we discussed this before, were you saying that you wanted rmq.all to **not** include the root view, but a filtered view `rmq(:stuff).all` should include `:stuff` and all the children, grandchildren etc?
